src/window.rs

```diff
@@ -0,0 +1,177 @@
+use core::marker::PhantomData;
+use core::f64;
+
+use {Sample, FloatSample};
+use signal;
+use signal::{Phase, ConstHz};
+use conv::{FromSample, ToSample};
+use frame::Frame;
+
+// Using types instead of enum allows for static dispatch
+pub trait Type<S: Sample> {
+    fn at_phase(phase: S) -> S;
+}
+
+pub struct Hanning;
+
+pub struct Rectangle;
+
+pub struct Window<F, WT> 
+    where F: Frame,
+          WT: Type<F::Sample>
+{
+    pub phase: Phase<ConstHz>,
+    ftype: PhantomData<F>,
+    wttype: PhantomData<WT>
+}
+
+pub struct WindowedFrame<'a, F, WT> 
+    where F: 'a + Frame,
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub data: &'a [F],
+    pub window: Window<F, WT>,
+    idx: usize
+}
+
+/// Windower takes a long slice of data and generates an iterator over its frames
+pub struct Windower<'a, F, WT> 
+    where F: 'a + Frame, 
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub bin: usize,
+    pub hop: usize,
+    pub idx: usize,
+    pub data: &'a [F],
+    wttype: PhantomData<WT>
+}
+
+impl<S: Sample + ToSample<f64> + FromSample<f64>> Type<S> for Hanning {
+    fn at_phase(phase: S) -> S {
+        let v = phase.to_sample::<f64>() * f64::consts::PI * 2.;
```

Comment:
> I was under the impression that to_sample::<f64> would normalize the i16 value to between -1 and 1. 

This is correct, so it might just be that my understanding of how this works is wrong! I'll leave it for now and do some more reading :+1: